app/controllers/application_controller.rb

```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    if session[:user_id]
+      return User.find(session[:user_id])
+    else
+      return nil
```

Comment:
You very rarely need to return nil in ruby - you could make this method just
```
def current_user
  User.find_by(id: session[:user_id]) if session[:user_id]
end
```

I prefer find_by over find here as find will raise an error if it fails, whereas for us returning nil is good enough. In the case where there is no session[:user_id] the line won't execute and the method will implicitly return nil.